client/handle_http.go

```diff
@@ -2256,6 +2243,82 @@ func runPut(request *http.Request, responseChan chan<- *http.Response, errorChan
 
 }
 
+// This function queries the director with a PROPFIND to attempt to get the 'dirListHost'. If a propfind is not allowed on the director
```

Comment:
I don't think we need to do a `PROPFIND` to get this information. Can you replace it with a `HEAD`?

Reply:
I'm not sure if a HEAD request would work, we do the PROPFIND to tell the director to redirect us straight to the origin, if we do a regular HEAD request, we might get sent to a cache. I could be wrong about this though. This helper function includes all the changes I recently made to directory listings, just made to be a helper function so other functions can access it.